processor/src/main/java/org/mapstruct/ap/internal/model/AbstractBaseBuilder.java

```diff
@@ -60,7 +60,8 @@ boolean canGenerateAutoSubMappingBetween(Type sourceType, Type targetType) {
     }
 
     private boolean isDisableSubMappingMethodsGeneration() {
-        MapperConfiguration configuration = MapperConfiguration.getInstanceOn( ctx.getMapperTypeElement() );
+        MapperConfiguration configuration = MapperConfiguration.getInstanceOn(
```

Comment:
I'm not so happy with the amount of `getInstanceOn` that we do and the distributed checking of the config. Shouldn't we treat it more like we treat other annotations: have a class doing the checking in our `org.mapstruct.ap.internal.model.source` package?

Reply:
Maybe I'm wrong, but unfortunatelly no. We haven't class in org.mapstruct.ap.internal.model.source for mapper configuration. We always use MapperConfiguration.getInstanceOn eight times. Sometimes I think, it's too early to do this task with checking. Firstly we should refactor, maybe. What do you think?